images/router/haproxy/conf/haproxy-config.template

```diff
@@ -167,18 +167,18 @@ frontend public
   tcp-request content accept if HTTP
 
   # check if we need to redirect/force using https.
-  acl secure_redirect base,map_reg(/var/lib/haproxy/conf/os_route_http_redirect.map) -m found
+  acl secure_redirect req.hdr(host),lower,map_reg(/var/lib/haproxy/conf/os_route_http_redirect.map) -m found
```

Comment:
`base` is more than `req.hdr(host)`; namely, it's the host plus path.  You're breaking routes with paths.